### PR TITLE
Table fmt

### PR DESCRIPTION
**Pull request by ali-rostami**
Hi
Based on the `lv_label_set_text_fmt` function, I tried to write a function for tables too.

### CODE REVIEW

**Comment by embeddedt** [comment]
```
src/lv_widgets/lv_table.c: In function ‘lv_table_set_cell_value_fmt’:
src/lv_widgets/lv_table.c:233:17: error: implicit declaration of function ‘lv_vsnprintf’; did you mean ‘lv_init’? [-Werror=implicit-function-declaration]
  uint32_t len = lv_vsnprintf(NULL, 0, fmt, ap);
```

File: src/lv_widgets/lv_table.c [hunk 1]
```diff
@@ -179,6 +179,101 @@ void lv_table_set_cell_value(lv_obj_t * table, uint16_t row, uint16_t col, const
...
+    _lv_mem_buf_release(raw_txt);
+#else
+    ext->cell_data[cell] = lv_mem_realloc(ext->cell_data[cell], len + 2); /*+1: trailing '\0; +1: format byte*/
+    LV_ASSERT_MEM(ext->cell_data[cell]);
+    if(ext->cell_data[cell] == NULL) {
+        va_end(ap2);
+        return;
+    }
+
+    ext->cell_data[cell][len - 1] = 0; /* Ensure NULL termination */
```
Comment (embeddedt) [inline src/lv_widgets/lv_table.c]: Is this the right byte to set to 0? Suppose `len` were 3 - I think the memory layout would look like this:

```
0 - format byte
1 - string char
2 - string char
3 - string char
4 - null byte
```

So I think `len+1` would be the correct byte to zero here.

Comment (ali-rostami) [reply src/lv_widgets/lv_table.c]: Yes, you are right, thanks.

**Comment by kisvegabor** [comment]
Hi,

looks good. It seems only an `#include "../lv-misc/lv_printf.h"` is missing to make `lv_vsnprintf` visible.

**Comment by kisvegabor** [comment]
Thanks!

CI still shows this error:
```
/home/runner/work/lvgl/lvgl/tests/../../lvgl/src/lv_widgets/lv_table.c: In function ‘lv_table_set_cell_value_fmt’:
/home/runner/work/lvgl/lvgl/tests/../../lvgl/src/lv_widgets/lv_table.c:220:55: error: ‘txt’ undeclared (first use in this function); did you mean ‘ext’?
             format.s.align = _lv_bidi_detect_base_dir(txt);
                                                       ^~~
                                                       ext
```

**Comment by ali-rostami** [comment]
> CI still shows this error:

Yeah, my mistake, fixed now.

**Review by embeddedt — APPROVED** [review]

**Comment by kisvegabor** [comment]
Merged to `dev`, thank you!

**Pull request merged**